lib/mongoid/extensions/range.rb

```diff
@@ -48,7 +48,14 @@ module ClassMethods
         #
         # @note Ruby 2.6 and lower do not support endless ranges that Ruby 2.7+ support.
         def demongoize(object)
-          object.nil? ? nil : ::Range.new(object["min"], object["max"], object["exclude_end"])
+          return if object.nil?
+          if object.is_a?(Hash)
+            hash = object.stringify_keys
+            hash.slice!('min', 'max', 'exclude_end')
+            unless hash.blank?
+              ::Range.new(hash["min"], hash["max"], hash["exclude_end"])
```

Comment:
And what about
```
::Range.new(hash[:min] || hash["min"], hash[:max] || hash["max"], hash[:exclude_end] || hash["exclude_end"])
```
and drop the `stringify_keys` and `slice!` calls?

Reply:
I think the `slice!` call is necessary so that we can check `blank?`. If we remove this then if someone passes in a bogus hash we'll end up initializing a `nil..nil` range. Plus, I don't think slice is that work intensive, it should just index into the hash three times no?

Reply:
I have removed `stringify_keys`

Reply:
I see. For input from DB which will have string keys (either via Hash or BSON::Document) key stringification is not needed.

Reply:
You need to take into account the symbol keys for this situation I believe: 
```
(byebug) u = User.new
#<User _id: 62cef5ee3282a407c8ee85ad, name: nil, r: nil, reg: nil, s: nil, time: nil>
(byebug) u.r = {min: 1, max: 2}
{:min=>1, :max=>2}
(byebug) u.r
1..2
(byebug) u.r_before_type_cast
{:min=>1, :max=>2}
```